include/Gaffer/SplinePlug.h

```diff
@@ -94,12 +94,14 @@ struct GAFFER_API SplineDefinition
 	// function would be the only way to setup one.
 	T spline() const;
 
-	// Trims off duplicated ends points after filling the points with data from
-	// a curve representation like IECore::Spline that adds duplicated
-	// end point values in order for evaluation of higher order interpolations to reach the end points.
+	// Remove start and end points with duplicated X values.
 	//
-	// Does not deal with unnecessary duplication of end points on constant and linear splines
-	// ( this is an OSL convention which is dealt with separately ).
+	// Sources of spline data may or may not contain duplicated end points for a variety of reasons
+	// ( such as IECore::Splineff having duplicated end points so that spline evaluation will reach
+	// the final value, or OSL have duplicated end points even for constant and linear splines ).
+	//
+	// The spline UI's that Gaffer uses to interact with splines don't support duplicated end
+	// points well, so regardless of why they are there, this function will remove them.
```

Comment:
I think it'd be worth adding some tests that directly exercise this function, rather than only testing via OSLShader, particularly as we're planning to move this class (but not OSLShader) to Cortex.